File: apps/mobile-legacy-rn/src/__tests__/auth.test.ts
import { useAuthStore } from '@/stores/authStore';
import { ROLES } from '@servis/shared';

describe('kimlik doğrulama (mock)', () => {
  beforeEach(async () => {
    await useAuthStore.getState().logout();
  });

  it('doğru demo bilgileriyle giriş yapılır ve rol atanır', async () => {
    await useAuthStore.getState().login('sofor@demo.com', 'Demo123!');
    const { user, status } = useAuthStore.getState();
    expect(status).toBe('authenticated');
    expect(user?.role).toBe(ROLES.DRIVER);
    expect(user?.email).toBe('sofor@demo.com');
  });

  it('hatalı şifre girişi reddeder', async () => {
    await expect(useAuthStore.getState().login('yolcu@demo.com', 'yanlis')).rejects.toThrow();
    expect(useAuthStore.getState().status).toBe('unauthenticated');
  });

  it('çıkış yapınca oturum temizlenir', async () => {
    await useAuthStore.getState().login('yonetici@demo.com', 'Demo123!');
    await useAuthStore.getState().logout();
    expect(useAuthStore.getState().user).toBeNull();
  });
});

File: apps/mobile/test/auth_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:servis_takip/core/constants/roles.dart';
import 'package:servis_takip/data/services/auth_service.dart';

void main() {
  final service = MockAuthService();

  test('doğru demo bilgileriyle giriş yapılır ve rol atanır', () async {
    final result = await service.login('sofor@demo.com', 'Demo123!');
    expect(result.user.role, Role.driver);
    expect(result.user.email, 'sofor@demo.com');
    expect(result.tokens.accessToken, isNotEmpty);
  });

  test('hatalı şifre girişi reddeder', () {
    expect(() => service.login('yolcu@demo.com', 'yanlis'), throwsException);
  });

  test('tanımsız e-posta reddeder', () {
    expect(() => service.login('yok@demo.com', 'Demo123!'), throwsException);
  });

  test('her rol için demo hesap girişi çalışır', () async {
    final emails = {
      'superadmin@demo.com': Role.superAdmin,
      'yonetici@demo.com': Role.companyAdmin,
      'sofor@demo.com': Role.driver,
      'yolcu@demo.com': Role.passenger,
    };
    for (final entry in emails.entries) {
      final result = await service.login(entry.key, 'Demo123!');
      expect(result.user.role, entry.value);
    }
  });
}
